config/initializers/rack_attack.rb

```diff
@@ -0,0 +1,101 @@
+  #
+  # Note: If you're serving assets through rack, those requests may be
+  # counted by rack-attack and this throttle may be activated too
+  # quickly. If so, enable the condition to exclude them from tracking.
+
+  # Throttle all requests by IP (60rpm)
+  #
+  # Key: "rack::attack:#{Time.now.to_i/:period}:req/ip:#{req.remote_ip}"
+  throttle('req/ip', limit: 300, period: 5.minutes) do |req|
+    req.remote_ip # unless req.path.start_with?('/assets')
+  end
+
+  ### Prevent Brute-Force Login Attacks ###
+
+  # The most common brute-force login attack is a brute-force password
+  # attack where an attacker simply tries a large number of emails and
+  # passwords to see if any credentials match.
+  #
+  # Another common method of attack is to use a swarm of computers with
+  # different IPs to try brute-forcing a password for a specific account.
+
+  # Throttle POST requests to /users/sign_in by IP address
+  #
+  # Key: "rack::attack:#{Time.now.to_i/:period}:logins/ip:#{req.remote_ip}"
+  throttle('logins/ip', limit: 5, period: 20.seconds) do |req|
+    if req.path == '/users/sign_in' && req.post?
+      req.remote_ip
+    end
+  end
+
+  # Throttle POST requests to /users/sign_in by email param
+  #
+  # Key: "rack::attack:#{Time.now.to_i/:period}:logins/email:#{normalized_email}"
+  #
+  # Note: This creates a problem where a malicious user could intentionally
+  # throttle logins for another user and force their login requests to be
+  # denied, but that's not very common and shouldn't happen to you. (Knock
+  # on wood!)
+  throttle('logins/email', limit: 5, period: 20.seconds) do |req|
+    if req.path == '/users/sign_in' && req.post?
+      if (req.params['user'].to_s.size > 0) and (req.params['user']['email'].to_s.size > 0)
+        # Normalize the email, using the same logic as your authentication process, to
+        # protect against rate limit bypasses. Return the normalized email if present, nil otherwise.
+        req.params['user']['email'].to_s.downcase.gsub(/\s+/, "").presence
+      end
+    end
+  end
+
+  ### Custom Throttle Response ###
+
+  # By default, Rack::Attack returns an HTTP 429 for throttled responses,
+  # which is just fine.
+  #
+  # If you want to return 503 so that the attacker might be fooled into
+  # believing that they've successfully broken your app (or you just want to
+  # customize the response), then uncomment these lines.
+  # self.throttled_response = lambda do |env|
+  #  [ 503,  # status
+  #    {},   # headers
+  #    ['']] # body
+  # end
+
+  # Block any IPs that makes requests for /etc/passwd, WP pages or any php. 
+  blocklist("fail2ban") do |req|
+    Rack::Attack::Fail2Ban.filter("fail2ban-#{req.remote_ip}", maxretry: 1, findtime: 1.day, bantime: 1.day) do
+      CGI.unescape(req.query_string) =~ %r{/etc/passwd} ||
+        req.path.include?("/etc/passwd") ||
+        req.path.include?("wp-admin") ||
+        req.path.include?("wp-login") ||
```

Comment:
might be worth adding `wp-content` here too along with a wildcard for *.pdf/.mp3 etc - I remember seeing those in a lot of logs a few years ago.